Userland/Libraries/LibC/stdio.cpp

```diff
@@ -591,8 +591,14 @@ int feof(FILE* stream)
 int fflush(FILE* stream)
 {
     if (!stream) {
-        dbgln("FIXME: fflush(nullptr) should flush all open streams");
-        return 0;
+        int rc = 0;
+        __pthread_mutex_lock(&s_open_streams_lock);
+        for (auto& file : *s_open_streams) {
+            ScopedFileLock lock(&file);
+            rc = file.flush() ? rc : EOF;
```

Comment:
Do we need to set the `EOF` indicator of the stream here as well, or is this done for free inside `::flush()`?

Reply:
This is already done here: https://github.com/SerenityOS/serenity/blob/6e37487477ae3059350a32baa337e376337ad752/Userland/Libraries/LibC/stdio.cpp#L103

Reply:
The error code is set inside `FILE::do_write()`.